server/src/main/java/org/apache/druid/server/coordinator/balancer/BalancerStrategyFactory.java

```diff
@@ -26,7 +26,7 @@
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "strategy", defaultImpl = CostBalancerStrategyFactory.class)
 @JsonSubTypes(value = {
     @JsonSubTypes.Type(name = "cost", value = CostBalancerStrategyFactory.class),
-    @JsonSubTypes.Type(name = "cachingCost", value = CachingCostBalancerStrategyFactory.class),
```

Comment:
CachingCostBalancerStrategyFactory should be removed from code.

Reply:
It is being used in some coordinator simulations and I have kept it around for the time being for some benchmarking and correctness comparison with upcoming strategies. It will most likely be removed before the next release of Apache Druid.

Reply:
Added javadoc to this effect.